From 5e83a3c2eaae3f4b3cd7c970c0ea78e4b93bf198 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Mon, 24 May 2021 18:05:54 +0300
Subject: [PATCH] [SW-43147] i2cslave: support i2c speed command

Support i2c slave direct command '101' which returns
the speed of the i2c slave.

Change-Id: Ia9301cfb27d5f83c843783ea7c924ea086d27f9d
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/preboot/module_i2c_slave.c b/preboot/module_i2c_slave.c
index 37f1a20..942c2ca 100644
--- a/preboot/module_i2c_slave.c
+++ b/preboot/module_i2c_slave.c
@@ -118,6 +118,9 @@
 	case HL_INDIRECT_CTRL_STS_DATA_3:
 		ret = i2csi_ctrl_handle_status(cmd);
 		break;
+	case HL_MODULE_I2C_FREQ:
+		ret = HL_SLAVE_FREQ_400KHZ;
+		break;
 	default:
 		if (HL_I2C_EVENT_IS_PEC(cmd))
 			ret = i2cslave_pec;
